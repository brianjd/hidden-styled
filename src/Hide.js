import styled from 'styled-components'

const breakpoints = {
  xs: '@media screen and (max-width: 40em)',
  sm: '@media screen and (min-width: 40em) and (max-width: 52em)',
  md: '@media screen and (min-width: 52em) and (max-width: 64em)',
  lg: '@media screen and (min-width: 64em)',
}

const hidden = key => props => props[key] ? {
  [breakpoints[key]]: {
    display: 'none'
  }
} : null

const xs = hidden('xs')
const sm = hidden('sm')
const md = hidden('md')
const lg = hidden('lg')

const Hide = styled([],
  xs,
  sm,
  md,
  lg
)

export default Hide
