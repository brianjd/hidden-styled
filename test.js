import test from 'ava'
import { ThemeProvider } from 'styled-components'
import { createElement as h } from 'react'
import { create as render } from 'react-test-renderer'
import Hide, {
  breakpoints,
  hidden,
  xs,
  sm,
  md,
  lg
} from './src/Hide'

test('renders', t => {
  const json = render(h(Hide)).toJSON()
  t.snapshot(json)
})

test('renders with xs', t => {
  const json = render(h(Hide, { xs: true })).toJSON()
  t.snapshot(json)
})

test('renders with sm', t => {
  const json = render(h(Hide, { sm: true })).toJSON()
  t.snapshot(json)
})

test('renders with md', t => {
  const json = render(h(Hide, { md: true })).toJSON()
  t.snapshot(json)
})

test('renders with lg', t => {
  const json = render(h(Hide, { lg: true })).toJSON()
  t.snapshot(json)
})

test('hidden returns a function', t => {
  t.is(typeof hidden('foo'), 'function')
})

test('hidden function returns a style object', t => {
  const a = hidden('sm')({ sm: true })
  t.is(typeof a, 'object')
  t.deepEqual(a, {
    [breakpoints.sm]: {
      display: 'none'
    }
  })
})

test('xs returns a style object', t => {
  const a = xs({ xs: true })
  t.deepEqual(a, {
    [breakpoints.xs]: {
      display: 'none'
    }
  })
})

test('sm returns a style object', t => {
  const a = sm({ sm: true })
  t.deepEqual(a, {
    [breakpoints.sm]: {
      display: 'none'
    }
  })
})

test('md returns a style object', t => {
  const a = md({ md: true })
  t.deepEqual(a, {
    [breakpoints.md]: {
      display: 'none'
    }
  })
})

test('lg returns a style object', t => {
  const a = lg({ lg: true })
  t.deepEqual(a, {
    [breakpoints.lg]: {
      display: 'none'
    }
  })
})

test.todo('breakpoints can be configured with theme')
