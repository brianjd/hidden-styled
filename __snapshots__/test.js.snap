// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`renders with lg 1`] = `
<div
  className="sc-bdVaJa hKJEBu"
/>
`;

exports[`renders with md 1`] = `
<div
  className="sc-bdVaJa gLCTHw"
/>
`;

exports[`renders with sm 1`] = `
<div
  className="sc-bdVaJa bEMGDB"
/>
`;

exports[`renders with xs 1`] = `
<div
  className="sc-bdVaJa kBXXYC"
/>
`;
